src/datasets/bert_dataset_sl.py

```diff
@@ -0,0 +1,93 @@
+from collections import defaultdict
+from typing import List, Dict, Union
+
+import torch
+import torch.nn.functional as F
+from transformers import BertTokenizer
+
+from src.datasets.sized_collated_dataset import SizedCollatedDataset
+from src.utils.utils import clean_text
+
+
+class BERTDatasetSL(SizedCollatedDataset[Union[List[List[Union[str, int]]], int]]):
```

Comment:
`Union[List[List[Union[str, int]]], int]` is pretty hard to read and understand 😅
You can add your own alias, e.g. `SL_DATASET_SAMPLE=Union[List[List[Union[str, int]]], int]`